src/components/DataTable/TableBatchActions.js

```diff
@@ -0,0 +1,62 @@
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const translationKeys = {
+  'carbon.table.batch.items.selected': 'items selected',
+  'carbon.table.batch.item.selected': 'item selected',
+};
+
+const translateWithId = id => translationKeys[id];
+
+const TableBatchActions = ({
+  className,
+  children,
+  shouldShowBatchActions,
+  totalSelected,
+  onCancel,
+  translateWithId: t,
+  ...rest
+}) => {
+  const batchActionsClasses = cx(
+    {
+      'bx--batch-actions': true,
+      'bx--batch-actions--active': shouldShowBatchActions,
+    },
+    className
+  );
+
+  return (
+    <div {...rest} className={batchActionsClasses}>
+      {children}
+      <div className="bx--batch-summary">
+        <p className="bx--batch-summary__para">
+          <span>{totalSelected}</span>{' '}
+          {totalSelected > 1
+            ? t('carbon.table.batch.items.selected')
+            : t('carbon.table.batch.item.selected')}
+        </p>
+        <button className="bx--batch-summary__cancel" onClick={onCancel}>
+          Cancel
+        </button>
+      </div>
+    </div>
+  );
+};
+
+TableBatchActions.translationKeys = Object.keys(translationKeys);
+
+TableBatchActions.propTypes = {
+  children: PropTypes.node,
```

Comment:
Let's document `propTypes` here! Seems that some other sub-components misses `propTypes` descriptions.

Reply:
I'll add them in, not for children and className though. Sound good?

Reply:
Yes it makes sense :+1: